Dashboard/app/js/lib/components/devices/AssignmentsEditView/screens/AssignDatapoints/SearchDatapoints.jsx

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable import/no-unresolved */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Checkbox from 'akvo-flow/components/reusable/Checkbox';
+import AssignmentContext from '../../assignment-context';
+
+export default class SearchDatapoints extends React.Component {
+  state = {
+    selectedDatapoints: [],
+  };
+
+  onSearch = e => {
+    e.preventDefault();
+
+    const searchKey = e.target.searchDatapoints.value;
+    this.context.actions.findDatapoints(searchKey);
+  };
+
+  onSelectDatapoint = (id, checked) => {
+    const { selectedDatapoints } = this.state;
+    let newSelectedDatapoints = [];
+
+    if (checked) {
+      newSelectedDatapoints = selectedDatapoints.concat(id);
+    } else {
+      newSelectedDatapoints = selectedDatapoints.filter(
+        device => device !== id
+      );
+    }
+
+    this.setState({ selectedDatapoints: newSelectedDatapoints });
+  };
+
+  addToAssignment = () => {
+    const { selectedDatapoints } = this.state;
+    const { datapoints } = this.context.data;
+    const { addDatapointsToAssignment } = this.context.actions;
+
+    const formattedSelectedDps = selectedDatapoints.map(sDp =>
```

Comment:
its not clear to me why the _formatted_ here

Reply:
So from what I can gather what you are calling `formattedSelectedDps` are the actual data points which is an array of objects whereas the `selectedDatapoints` is an array of numbers that represent the ids of the datapoints.

Reply:
Yes. But I have to clarify that too